### PR TITLE
[trivial] Cleanup maxuploadtarget (doc & log)

### LINKED ISSUE #6890: Add -maxuploadtarget to 0.12 release-notes

### PR DESCRIPTION
**Pull request by maflcko**
@jonasschnelli this fixes  #6890.

### CODE REVIEW

File: qa/rpc-tests/maxuploadtarget.py [hunk 1]
```diff
@@ -192,7 +192,8 @@ def run_test(self):
         getdata_request.inv.append(CInv(2, big_old_block))
 
         max_bytes_per_day = 200*1024*1024
-        max_bytes_available = max_bytes_per_day - 144*1000000
+        daily_buffer = 144 * 1000000 # 144 * MAX_BLOCK_SIZE
```
Comment (jonasschnelli) [inline qa/rpc-tests/maxuploadtarget.py]: Hmm.... i think either we take in the `MAX_BLOCK_SIZE` constant (controversial) or we drop the comment.

**Comment by jonasschnelli** [comment]
utACK. Thanks!

I think we should extend the documentation about -maxuploadtarget. How it actually works under the hood (short, 1-2 sentence). I can do this after this has been merged.

File: qa/rpc-tests/maxuploadtarget.py (hunk 1 above)
Comment (MarcoFalke) [inline qa/rpc-tests/maxuploadtarget.py]: Is this exposed via rpc? I always like to do `grep -r 'CONSTANT' bitcoin/` when touching a constant to see it's usage.

**Comment by jonasschnelli** [comment]
Two commits that could make sense for this PR (2nd one maybe independent):

- db504eb6ca6c85db13bb7168c2c5d48840bb97d6 ([doc] extend -maxuploadtarget release notes)
- f6b96ce09ec827bb9be0fd926b199009dd08bfd4 ([doc] add documentation how to reduce traffic)

Both are written in my swenglish (swiss english). Feel free (or obligated) to correct and rewrite everything.

File: src/net.cpp [hunk 2]
```diff
@@ -2108,8 +2108,8 @@ void CNode::SetMaxOutboundTarget(uint64_t limit)
     uint64_t recommendedMinimum = (nMaxOutboundTimeframe / 600) * MAX_BLOCK_SIZE;
     nMaxOutboundLimit = limit;
 
-    if (limit < recommendedMinimum)
-        LogPrintf("Max outbound target is very small (%s) and will be overshot. Recommended minimum is %s\n.", nMaxOutboundLimit, recommendedMinimum);
+    if (limit > 0 && limit < recommendedMinimum)
```
Comment (jonasschnelli) [inline src/net.cpp]: Nice catch. Thanks!

**Comment by jonasschnelli** [comment]
utACK

**Comment by MarcoFalke** [comment]
Looks ready to merge!?

File: doc/release-notes.md [hunk 3]
```diff
@@ -151,6 +151,23 @@ mining with the getblocktemplate protocol to a pool: this will affect you at
...
+
+It is now possible to reduce the total upload traffic via the `-maxuploadtarget`
+parameter. This is *not* a hard limit but a threshold to minimize the outbound
+traffic. When the limit is about to be reached, the uploaded data is cut by not
+serving historic blocks (blocks older than one week).
+Moreover, any SPV peer is disconnected when they request a filtered block.
+
+This option can be specified in MiB per day and is turned off by default
+(`-maxuploadtarget=0`).
+The recommended minimum is 144 * MAX_BLOCK_SIZE (currently 144MB) per day.
```
Comment (jtimon) [inline doc/release-notes.md]: Maybe a link to https://github.com/jonasschnelli/bitcoin/blob/2015/11/doc_traffic/doc/reducetraffic.md (once it's merged) can be added here ? Something like "for more details on reducing traffic, see ...".

**Comment by jtimon** [comment]
utACK besides my little nit.

**Comment by MarcoFalke** [comment]
@jtimon Done.

**Comment by jtimon** [comment]
Great, re-utACK

**Pull request merged**